akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -9,136 +9,51 @@ import akka.util.Helpers
+import java.util.ArrayList
 
 /**
- * An Actor is given by the combination of a [[Behavior]] and a context in
- * which this behavior is executed. As per the Actor Model an Actor can perform
- * the following actions when processing a message:
- *
- *  - send a finite number of messages to other Actors it knows
- *  - create a finite number of Actors
- *  - designate the behavior for the next message
- *
- * In Akka the first capability is accessed by using the `!` or `tell` method
- * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
- * and the third is implicit in the signature of [[Behavior]] in that the next
- * behavior is always returned from the message processing logic.
- *
- * An `ActorContext` in addition provides access to the Actor’s own identity (“`self`”),
- * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
- * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ * This trait is not meant to be extended by user code. If you do so, you may
+ * lose binary compatibility.
  */
-trait ActorContext[T] {
-
-  /**
-   * The identity of this Actor, bound to the lifecycle of this Actor instance.
-   * An Actor with the same name that lives before or after this instance will
-   * have a different [[ActorRef]].
-   */
-  def self: ActorRef[T]
-
-  /**
-   * Return the mailbox capacity that was configured by the parent for this actor.
-   */
-  def mailboxCapacity: Int
-
-  /**
-   * The [[ActorSystem]] to which this Actor belongs.
-   */
-  def system: ActorSystem[Nothing]
-
-  /**
-   * The list of child Actors created by this Actor during its lifetime that
-   * are still alive, in no particular order.
-   */
-  def children: Iterable[ActorRef[Nothing]]
-
-  /**
-   * The named child Actor if it is alive.
-   */
-  def child(name: String): Option[ActorRef[Nothing]]
-
-  /**
-   * Create a child Actor from the given [[Props]] under a randomly chosen name.
-   * It is good practice to name Actors wherever practical.
-   */
-  def spawnAnonymous[U](behavior: Behavior[U], deployment: DeploymentConfig = EmptyDeploymentConfig): ActorRef[U]
-
-  /**
-   * Create a child Actor from the given [[Props]] and with the given name.
-   */
-  def spawn[U](behavior: Behavior[U], name: String, deployment: DeploymentConfig = EmptyDeploymentConfig): ActorRef[U]
-
-  /**
-   * Force the child Actor under the given name to terminate after it finishes
-   * processing its current message. Nothing happens if the ActorRef does not
-   * refer to a current child actor.
-   *
-   * @return whether the passed-in [[ActorRef]] points to a current child Actor
-   */
-  def stop(child: ActorRef[Nothing]): Boolean
+trait ActorContext[T] extends javadsl.ActorContext[T] with scaladsl.ActorContext[T] {
+  def getChild(name: String): Optional[ActorRef[Void]] =
+    child(name) match {
+      case Some(c) ⇒ Optional.of(c.upcast[Void])
+      case None    ⇒ Optional.empty()
+    }
```

Comment:
OptionConverters would try to convert `ActorRef[Nothing]` and the result of that is not terribly useful for Java users.